plugins/modules/zos_apf.py

```diff
@@ -0,0 +1,506 @@
+            data set recovery can be done by renaming it.
+        required: false
+        type: str
+  batch:
+    description:
+      - A list of dictionaries for adding/removing libraries
+      - This is mutually exclusive with C(dsname), C(volume), C(sms)
+      - Can be used with C(persistent)
+    type: list
+    elements: dict
+    required: false
+    suboptions:
+      dsname:
+        description:
+          - The name of z/OS data set (library) to be added/removed.
+        type: str
+        required: True
+        aliases: [ name, lib, library ]
+      volume:
+        description:
+          - The volume identifier for the volume containing the library
+            specified on the C(dsname) parameter, value should be one of the
+            following,
+            1. The volume serial number
+            2. Six asterisks (******), indicating that the system is to use the
+            volume serial number of the current system residence (SYSRES)
+            volume.
+            3. *MCAT*, indicating that the system is to use the volume serial
+            number of the volume containing the master catalog.
+          - If C(volume) is not specified, C(dsname) has to be cataloged.
+        required: False
+        type: str
+      sms:
+        description:
+          - Indicates that the library specified on the C(dsname) parameter is
+            managed by the storage management subsystem (SMS), and therefore no
+            volume is associated with the library.
+          - If true C(volume) will be ignored.
+        required: False
+        type: bool
+        default: False
+'''
+
+
+EXAMPLES = r'''
+- name: Add a library to APF list
+  zos_apf:
+    dsname: SOME.SEQUENTIAL.DATASET
+    volume: T12345
+- name: Add a library (cataloged) to APF list and persistence
+  zos_apf:
+    dsname: SOME.SEQUENTIAL.DATASET
+    force_dynamic: True
+    persistent:
+      persistds: SOME.PARTITIONED.DATASET(MEM)
+- name: Remove a library from APF list and persistence
+  zos_apf:
+    state: absent
+    dsname: SOME.SEQUENTIAL.DATASET
+    volume: T12345
+    persistent:
+      persistds: SOME.PARTITIONED.DATASET(MEM)
+- name: Use batch to add a set of libraries to APF list and persistence and custom marker
+  zos_apf:
+    persistent:
+      persistds: SOME.PARTITIONED.DATASET(MEM)
+      marker: "/* {mark} PROG001 USR0010 */"
+    batch:
+      - dsname: SOME.SEQ.DS1
+      - dsname: SOME.SEQ.DS2
+        sms: True
+      - dsname: SOME.SEQ.DS3
+        volume: T12345
+- name: Print APF list matching dsname pattern or volume serial number
+  zos_apf:
+    operation: list
+    dsname: SOME.SEQ.*
+    volume: T12345
+- name: Set APF list format to STATIC
+  zos_apf:
+    operation: set_static
+'''
+
+
+RETURN = r'''
+stdout:
+  description:
+    - The stdout from ZOAU apfadm. Output varies based on type of operation.
+    - state> stdout of executed the operator command (opercmd), "SETPROG" from ZOAU apfadm
```

Comment:
"- state> stdout of executed the operator command (opercmd), "SETPROG" from ZOAU apfadm" 
to
"- state> stdout of the executed operator command (opercmd), "SETPROG" from ZOAU command apfadm"

Reply:
done